src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -283,47 +304,49 @@ emit_common_simd_operations (TransformData *td, int id, int atype, int vector_si
 }
 
 static gboolean
-get_common_simd_info (MonoClass *vector_klass, MonoMethodSignature *csignature, MonoTypeEnum *atype, int *vector_size, int *arg_size, int *scalar_arg)
+get_common_simd_info (MonoClass *klass, MonoTypeEnum *atype, int *klass_size, int *arg_size)
```

Comment:
This file ended up needing a bit of a refactoring as there were some assumptions in place that don't hold when supporting additional intrinsics.

In particular, there are various intrinsics where:
* one of the SIMD types may not be generic at all (`Vector2`, `Vector3`, `Vector4`)
* multiple generic types exist (`As<TFrom, TTo>`)
* the return type may not be a 128-bit vector (`AsVector2`, `AsVector3`)

So, what I did here was I broke this `get_common_simd_info` method into two:
* `get_common_simd_info`
* `get_common_simd_scalar_arg`

The former now always gets the size of the input klass and secondly determines if it is a SIMD type and what the underlying element type is if so. While the latter identifies the first non-SIMD argument, if one exists.